### PR TITLE
Remove function signatures - Closes #4571

### LINKED ISSUE #4571: Remove function signatures
### Description
- Remove function signature or skeletons
- Add missing code comments if required
### Motivation
In order to unify the function signature standards across `lisk-sdk` and `lisk-core`, we want to remove all the function signature or skeletons which are partially implemented or sometimes not implemented at all.
This partial implementation leads to inconsistency in the overall experience.

- Missing description, not maintained
```
/**	
	 *	
	 * @param {Object} options	
	 * @param {Boolean} options.inTest	
	 * @param {string} options.logger	
	 * @param {string} options.sqlDirectory	
	 */
```
- Keeping skeleton which doesn't add much value
```
/**	
	 * Description of the function.	
	 *	
	 * @param {string} str	
	 * @returns {boolean}	
	 * @todo Add description for the function, the params and the return value	
	 */
```
- These and many more missing standards for function signature document leads to inconsistancy.

Note: Code commenting is still allowed and appreciated where it's required, let's make an effort to write functions which are more explicit in nature so that we do need these comments.

### Acceptance Criteria
- Remove function signature or skeletons
- Add missing code comments if required

### PR DESCRIPTION
**Pull request by ManuGowda**
### What was the problem?
In order 
to unify the function signature standards across lisk-sdk and lisk-core we want to remove all the function signature or skeletons which are partially implemented or sometimes not implemented at all.
This partial implementation leads to inconsistency in the overall experience.

Motivation Examples:
- Missing description, not maintained
```
/**	
	 *	
	 * @param {Object} options	
	 * @param {Boolean} options.inTest	
	 * @param {string} options.logger	
	 * @param {string} options.sqlDirectory	
	 */
```
- Keeping skeleton which doesn't add much value
```
/**	
	 * Description of the function.	
	 *	
	 * @param {string} str	
	 * @returns {boolean}	
	 * @todo Add description for the function, the params and the return value	
	 */
```
- These and many more missing standards for function signature document leads to inconsistancy.

Note: Code commenting is still allowed and appreciated where it's required, let's make an effort to write functions which are more explicit in nature so that we do need these comments.

### How did I solve it?
- Remove function signature or skeletons
- Add missing code comments if required
### How to manually test it?
Please verify if I am missing any function signatures to be removed.
### Review checklist

- [ ] The PR resolves #4571 
- [ ] All new code is covered with unit tests
- [ ] Relevant integration / functional tests are added
- [ ] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [ ] Documentation has been added/updated

### CODE REVIEW

**Comment by limiaspasdaniel** [comment]
Why have the storage function JSDoc comments been removed? IMO precisely those were the ones that were best implemented across all our codebase. I don't see the value of removing those.

**Comment by pablitovicente** [comment]
I think we should keep the objective of the functions removing only the parameters. Whenever we can refactor the code to use better naming we could remove the whole header for each function but right now I think we should keep them and in any case update the objective with proper descriptions.

**Comment by ManuGowda** [comment]
@pablitovicente I had a discussion with @limiaspasdaniel about this, maybe he can explain the rationale, for now, we also briefly discussed this in the standup and we will keep the code comments, and remove the function signature to keep it consistent across SDK. I understand its an opinionated topic, so we can touch base on this topic on Monday. If you see anything important code comment being removed in this PR please do leave the comment, I will address them. Thanks 👍

File: framework/test/mocha/unit/modules/http_api/controllers/delegates.js [hunk 1]
```diff
@@ -315,7 +315,7 @@ describe('delegates/api', () => {
 			try {
 				await __private.aggregateBlocksReward({ address: '1L' });
 			} catch (err) {
-				expect(err).to.deep.equal('Blocks#aggregateBlocksReward error');
+				expect(err.message).to.deep.equal('Blocks#aggregateBlocksReward error');
```
Comment (shuse2) [inline framework/test/mocha/unit/modules/http_api/controllers/delegates.js]: 🤔

Comment (ManuGowda) [reply framework/test/mocha/unit/modules/http_api/controllers/delegates.js]: before it was not throwing the error, it was throwing the message, any specific reason that we throw the message and not the error with message?

Comment (shuse2) [reply framework/test/mocha/unit/modules/http_api/controllers/delegates.js]: there was no specific message, but if we gonna change it it should be changed consistently in separate PR

Comment (ManuGowda) [reply framework/test/mocha/unit/modules/http_api/controllers/delegates.js]: agree.. I will revert it back.. and create a separate issue to address this

**Pull request merged**